justfile

```diff
@@ -60,11 +107,11 @@ euclid-wasm features='dummy_connector':
 # Run pre-commit checks
 precommit: fmt clippy
 
-hack_flags := '--workspace --each-feature --all-targets'
-
-# Check compilation of each cargo feature
-hack:
-    cargo hack check {{ hack_flags }}
+# Check compilation of v2 feature on base dependencies
+v2_intermediate_features := "merchant_account_v2,payment_v2"
+hack_v2:
+    cargo hack check  --feature-powerset --ignore-unknown-features --at-least-one-of "v2 " --include-features "v2" --include-features {{ v2_intermediate_features }} --package "hyperswitch_domain_models" --package "diesel_models" --package "api_models"
```

Comment:
Specifying `--feature-powerset` can grow exponentially as and when new features are added/included.